recipe/build.sh

```diff
@@ -12,6 +12,8 @@ cmake ${CMAKE_ARGS}\
   -DMI_BUILD_SHARED=ON \
   -DMI_BUILD_STATIC=OFF \
   -DMI_BUILD_OBJECT=OFF \
+  -DMI_OVERRIDE=OFF \
+  -DMI_LOCAL_DYNAMIC_TLS=ON \
```

Comment:
Changing the [Thread Local Storage (TLS) model](https://fuchsia.dev/fuchsia-src/development/kernel/threads/tls) from the default ([currently set to `initial-exec`](https://github.com/microsoft/mimalloc/blob/03020fbf81541651e24289d2f7033a772a50f480/CMakeLists.txt#L331)) to `local-dynamic` significantly changes the builds.

I wonder if it would make sense to introduce a build variant with a `track_feature` for selecting the TLS.

This would has the advantages of having the current main build unchanged whilst allowing other projects (like ArcticDB) to use another TLS by using a specification, e.g.:

```
mamba install "mimalloc=*=*local_dynamic_tls"
```

Reply:
Seems like a good choice.

Reply:
Would merging the PR as is break the ABI?

Reply:
Probably not. You could check by looking at the diff of `nm` outputs.

Reply:
It unfortunately changes the ABI by removing symbols.

After building on `main` and this branch:

```bash
# main (at 7980757)
mimalloc_shared_object_at_7980757=build_artifacts/linux-64/mimalloc-2.1.7-hac33072_0/pkg-mimalloc-2.1.7-hac33072_0/lib/libmimalloc.so.2.1

# this branch
mimalloc_shared_object_at_38a3781=build_artifacts/linux-64/mimalloc-2.1.7-hac33072_1/pkg-mimalloc-2.1.7-hac33072_1/lib/libmimalloc.so.2.1

diff <(nm --just-symbols $mimalloc_shared_object_at_7980757) <(nm --just-symbols $mimalloc_shared_object_at_38a3781)
```

```diff
4d3
< _aligned_malloc
7,8d5
< calloc
< cfree
30d26
< free
40,47d35
< __libc_calloc
< __libc_cfree
< __libc_free
< __libc_malloc
< __libc_memalign
< __libc_pvalloc
< __libc_realloc
< __libc_valloc
49,53d36
< malloc
< malloc_good_size
< malloc_size
< malloc_usable_size
< memalign
425,426d407
< __posix_memalign
< posix_memalign
433,437d413
< pvalloc
< realloc
< reallocarr
< reallocarray
< reallocf
445d420
< strndup
453a429
> __tls_get_addr@GLIBC_2.3
457,458d432
< valloc
< vfree
460,479d433
< _ZdaPv
< _ZdaPvm
< _ZdaPvmSt11align_val_t
< _ZdaPvRKSt9nothrow_t
< _ZdaPvSt11align_val_t
< _ZdaPvSt11align_val_tRKSt9nothrow_t
< _ZdlPv
< _ZdlPvm
< _ZdlPvmSt11align_val_t
< _ZdlPvRKSt9nothrow_t
< _ZdlPvSt11align_val_t
< _ZdlPvSt11align_val_tRKSt9nothrow_t
< _Znam
< _ZnamRKSt9nothrow_t
< _ZnamSt11align_val_t
< _ZnamSt11align_val_tRKSt9nothrow_t
< _Znwm
< _ZnwmRKSt9nothrow_t
< _ZnwmSt11align_val_t
< _ZnwmSt11align_val_tRKSt9nothrow_t
```